rtf_vis_tool/src/Components/DivGraph.js

```diff
@@ -0,0 +1,164 @@
+const DivGraph = (props) => {
+    const [arrowList, setArrowList] = useState([]);
+    const leftShift = 0;
+    const topShift = 0;
+    
+    const aquaBlue = '#2255e0';
+    const lightGray = '#dfe8e6';
+
+    const [showFS, setShowFS] = useState(false);
+    const [fs_json, setFS_JSON] = useState(null);
+    const [showMVO, setShowMVO] = useState(false);
+    const [mvo_json, setMVO_JSON] = useState(null);
+    const [da_json, setDA_JSON] = useState(null);
+    const [showDA_PC, setShowDA_PC] = useState(null);
+    const [showRSS, setShowRSS] = useState(false);
+    const [rotated_da_json, setRotatedDAJSON] = useState(null);
+
+    //render all directed edges on the graph
+    //save all the json resulting metrics in separate React variables
+    useEffect(() => {
+        var rawEdges = EdgeList
+        var xArrows_formatted = [];
+
+        for (var i = 0; i < rawEdges.length; i++) {
+            const pair = rawEdges[i];
+            xArrows_formatted.push(
+                <Xarrow
+                    start={pair[0]}
+                    end={pair[1]}
+                    color='gray'
+                    strokeWidth='1.5'
+                    path='straight'
+                />)
+        }
+
+        setArrowList(xArrows_formatted);
+        setFS_JSON(frontend_summary_json);
+        setMVO_JSON(multiview_optimizer_json);
+        setDA_JSON(data_association_json.points_3d);
+        setRotatedDAJSON(data_association_json.rotated_points_3d);
+    }, [])
+
+    //Function to defines the percent offset each div node has from the top and left side of the screen
+    //Used to make positioning of nodes more dynamic
+    function formatPercent(shift, percent) {
+        const str_percent = `${shift+percent}%`;
+        return str_percent
+    }
+
+    //Functions to toggle the display of various pop ups on the screen
+    //Like frontend metrics, multiview optimizer metrics, and data association point cloud
+    const toggleFrontEndSummaryDisplay = (bool) => {setShowFS(bool)};
+    const toggleMVOMetrics = (bool) => {setShowMVO(bool)};
+    const toggleDataAssoc_PointCloud = (bool) => {setShowDA_PC(bool)};
+    const toggleRotSummaryDisplay = (bool) => {setShowRSS(bool)};
+
+    return (
+        <div className="div_graph_container">
+            <div className="navbar">
+                <h2 className="gtsfm_header">GTSFM Computational Graph Visualizer</h2>
+            </div>
+
+            {/* Render popups only when the respective node is clicked*/}
+            {/* {showFS && <FrontendSummary json={fs_json} toggleFS={toggleFrontEndSummaryDisplay}/>} */}
+            {/* {showMVO && <MVOSummary json={mvo_json} toggleMVO={toggleMVOMetrics}/>} */}
+            {showDA_PC && <Data_Association_PC da_json={da_json} rotated_json={rotated_da_json} toggleDA_PC={toggleDataAssoc_PointCloud}/>}
+            {/* {showRSS && <RotSuccessSummary json={fs_json} toggleRSS={toggleRotSummaryDisplay}/>} */}
```

Comment:
some of this is commented out because they are attached to components which I have not yet added to this PR (for simplicity's sake). In the 3rd PR, I will add the components related to these and appropriately uncomment these lines.